### PR TITLE
minor fixes and additions

### PR DESCRIPTION
**Pull request by Slaw6820**
During my work with apitrace I have encountered a small number of issues that were addressed with the following changes to the source code.
Please review.

### CODE REVIEW

**Comment by okias** [comment]
Hello @Slaw6820 

thank you for the contribution, could you split the one commit into the separate changes?

**Comment by Slaw6820** [comment]
Sure. Does it mean that all proposed fixes are accepted?
Should I create multiple changes under one PR or separate PRs?

**Comment by okias** [comment]
> Sure. Does it mean that all proposed fixes are accepted? Should I create multiple changes under one PR or separate PRs?

LTGM, but I'm not an biggest authority here :) just commits within this MR :)

**Comment by Slaw6820** [comment]
@jrfonseca

File: helpers/glfeatures.hpp [hunk 1]
```diff
@@ -99,19 +99,19 @@ struct Profile {
...
 
     // Comparison operator, mainly for use in std::map
     inline bool
     operator < (const Profile & other) const {
-        return major < other.major ||
-               minor < other.minor ||
-               api < other.api ||
-               core < other.core ||
-               forwardCompatible < other.forwardCompatible;
+        return (major  < other.major) ||
```
Comment (jrfonseca) [inline helpers/glfeatures.hpp:110]: Good catch.

There's a more compact way to achieve the same:
```c++
        if (major != other.major) {
            return major < other.major;
        }
        if (minor != other.minor) {
            return minor < other.minor;
        }
        ...
````

File: frametrim/ft_opengl.cpp [hunk 2]
```diff
@@ -221,7 +221,16 @@ OpenGLImpl::skipDeleteObj(const trace::Call& call)
         map = &m_textures;
 
     if (!strcmp(call.name(), "glDeleteFramebuffers"))
-        map = &m_current_context->m_fbo;
+    {
```
Comment (jrfonseca) [inline frametrim/ft_opengl.cpp:224]: Code formatting should be
```c++
    if (condition) {
        ....
    }
````

File: retrace/retrace_swizzle.cpp [hunk 3]
```diff
@@ -32,6 +32,19 @@
 #include "retrace_swizzle.hpp"
 
 
+#ifdef assert
+#undef assert
+#endif
+
+
+#define assert(expression) \
```
Comment (jrfonseca) [inline retrace/retrace_swizzle.cpp:40]: Why is this necessary?

**Review by jrfonseca — COMMENTED** [review]
Sorry for the silence -- I was on vacation last week.

Thanks for the changes.  Comments inline.

Otherwise looks good.   I agree it would be good to break the changes according to main _modules_ (wrappers, retrace, etc.  The main reason is that one can git bisect and revert chunks
